src/library-authoring/create-container/CreateContainerModal.test.tsx

```diff
@@ -0,0 +1,232 @@
+import MockAdapter from 'axios-mock-adapter';
+import userEvent from '@testing-library/user-event';
+import {
+  render, screen, waitFor, initializeMocks,
+} from '../../testUtils';
+
+import { LibraryProvider } from '../common/context/LibraryContext';
+import CreateContainerModal from './CreateContainerModal';
+import AddContent from '../add-content/AddContent';
+import { getLibraryCollectionItemsApiUrl, getLibraryContainerChildrenApiUrl } from '../data/api';
+
+let axiosMock: MockAdapter;
+const mockNavigateTo = jest.fn();
+let mockShowToast: any;
+
+const mockUseLibraryRoutes = jest.fn().mockReturnValue({
+  navigateTo: mockNavigateTo,
+  insideCollection: true,
+  insideSection: false,
+  insideSubsection: false,
+});
+
+jest.mock('../routes', () => ({
+  useLibraryRoutes: () => mockUseLibraryRoutes(),
+}));
+
+const mockLibraryData = {
+  id: 'test-library',
+  title: 'Test Library',
+  description: 'A test library',
+  version: 1,
+  type: 'complex',
+  license: '',
+  canEditLibrary: true,
+  hasUnpublishedChanges: false,
+  hasUnpublishedDeletes: false,
+  numBlocks: 0,
+  lastPublished: null,
+  created: '2024-01-01T00:00:00Z',
+  updated: '2024-01-01T00:00:00Z',
+  publishedBy: null,
+  lastDraftCreated: null,
+  lastDraftCreatedBy: null,
+  readOnly: false,
+  allowLti: false,
+  allowPublicLearning: false,
+  allowPublicRead: false,
+  libraryBlockTypes: [],
+  learningPackage: null,
+};
+
+const mockBlockTypesData = [
+  { blockType: 'html', displayName: 'Text' },
+  { blockType: 'problem', displayName: 'Problem' },
+  { blockType: 'video', displayName: 'Video' },
+];
+
+const renderWithLibraryProvider = (libraryId: string, options: {
+  collectionId?: string;
+  containerId?: string;
+  path?: string;
+  params?: Record<string, string>;
+} = {}) => {
+  const {
+    collectionId, containerId, path, params,
+  } = options;
+
+  axiosMock.onGet(`/api/libraries/v2/${libraryId}/`).reply(200, mockLibraryData);
+  axiosMock.onGet(`/api/libraries/v2/${libraryId}/block_types/`).reply(200, mockBlockTypesData);
```

Comment:
You can use [mockBlockTypesMetadata](https://github.com/openedx/frontend-app-authoring/blob/b5fd147eec9bb0095260f152a3193445c112ae72/src/library-authoring/data/api.mocks.ts#L714C23-L714C45) to mock this call and data